packages/model-viewer/src/features/scene-graph.ts

```diff
@@ -283,6 +295,33 @@ export const SceneGraphMixin = <T extends Constructor<ModelViewerElementBase>>(
     [$onModelGraftMutation](_event: Event) {
       this[$needsRender]();
     }
+
+    /** @export */
+    async exportScene(options: {
+			binary: false,
+			trs: false,
+			onlyVisible: true,
+			truncateDrawRange: true,
+			embedImages: true,
+			maxTextureSize: 1024,
+			animations: [],
+			forcePowerOfTwoTextures: false,
+			includeCustomExtensions: false
+		}): Promise<Blob> {
+      
+      const { model } = this[$scene];
+      return new Promise<Blob>(async (resolve, reject) => {
+        if (model == null) { 
+          return reject( 'Model missing or not yet loaded' );
+        }
+        
+        var exporter = new GLTFExporter();
+        exporter.parse( model, ( gltf ) => {
+            return resolve(new Blob([options.binary ? gltf as Blob : JSON.stringify(gltf)], {type: options.binary ? 'application/octet-stream' : 'application/json'}));
```

Comment:
A glTF is usually several files; does this just get the json part, or the rest as well?

Reply:
Textures are included in the exported glTF as base64 encoded files